### DISCUSSION TITLE
how to scroll to first validation error message in livewire laravel

### DISCUSSION THREAD

**Discussion by kunalm7865** [livewire/livewire#2542] [4. General]
I am new in livewire. i am working on registration process of user. Register form is quite big and validations also working fine to me. The problem is when user submits the form validations came but user not able to see because submit button is at bottom and form is at top.

When i manually scroll to top error message is displaying. So what i want when user submits the form its will automatically go to first error message. Here is my code:

    <form wire:submit.prevent="userRegister">
	<div class="row">
		<div class="col-md-3">
			<label>First Name *</label>
		</div>
		<div class="col-md-9">
			<input type="text" wire:model="register.first_name" placeholder="Enter First Name" required>
			@error('register.first_name') <span class="text-danger">{{ $message }}</span> @enderror
		</div>
	</div>
	<div class="row">
		<div class="col-md-3">
			<label>Last Name *</label>
		</div>
		<div class="col-md-9">
			<input type="text" wire:model="register.last_name" placeholder="Enter Last Name">
			@error('register.last_name') <span class="text-danger">{{ $message }}</span> @enderror
		</div>
	</div>
	<div class="row">
		<div class="col-md-3">
			<label>Test 1 Name *</label>
		</div>
		<div class="col-md-9">
			<input type="text" wire:model="register.test1_name" placeholder="Enter Last Name">
		</div>
	</div>
	<div class="row">
		<div class="col-md-3">
			<label>Test 2 Name *</label>
		</div>
		<div class="col-md-9">
			<input type="text" wire:model="register.test2_name" placeholder="Enter Last Name">
		</div>
	</div>
	<div class="row">
		<div class="col-md-3">
			<label>Test 3 Name *</label>
		</div>
		<div class="col-md-9">
			<input type="text" wire:model="register.test3_name" placeholder="Enter Last Name">
		</div>
	</div>
	<div class="row">
		<div class="col-md-3"></div>
		<div class="col-md-9">
			
			<input wire:click="userRegister" type="submit" value="Submit" class="edt-sv">
		</div>
	</div>
    </form>

My Register.php

    <?php

    namespace App\Http\Livewire;

    use Livewire\Component;
    class Register extends Component
    {
	public $register;
	protected $rules = [
        'register.first_name' => 'bail|required|max:50',
        'register.last_name' => 'bail|required|max:50',
    ];

    protected $messages = [
	    'register.first_name.required' => 'Please enter first name',
	];

    public function userRegister(){
    	$this->validate();
    }

I want when user submits the form it will immediately scroll to first error message. Currently my validations work perfects for me. Do i need to use alpine js? for this.

**Comment by joshhanley** [comment]
@kunalm7865 I'd suggest searching through Livewire issues and discussions first, as generally there is an answer somewhere.

I've just done a quick search for "scroll validation" on the issues and found this one https://github.com/livewire/livewire/issues/1415

If you look at the last couple of comments there is a potential solution there. 

Give that a try and if you have troubles implementing it, then either post a question on that issue or back here 🙂

Generally you will get a better response if you attempt to solve the issue yourself, and then post up what you have tried and whether there are any errors. It's much easier to help fix some small issue with your code that it is to try and work out how to do the thing from scratch (as I've personally never done scroll to validation message).

Hope this helps!
